Bottom: cd55feeac2aea07a9d2b47e439a6899364a15c24
Top:    8a7c16d64a7b5bd3d9347428bb365d347d824b30
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:22:06 +0000

Refresh of fixup-board_outline_poly-to

---

diff --git a/src/polygon.c b/src/polygon.c
index e780ded..fc83944 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2092,10 +2092,8 @@ POLYAREA *board_outline_poly (bool include_holes)
         }
     }
 
-  if (!found_outline) {
+  if (!found_outline)
     printf ("Didn't find outline\n");
-    return whole_world;
-  }
 
   /* Do stuff to turn the outline layer into a polygon */
 
@@ -2127,8 +2125,11 @@ POLYAREA *board_outline_poly (bool include_holes)
   region.X2 = PCB->MaxWidth;
   region.Y2 = PCB->MaxHeight;
 
-  r_search (Layer->line_tree, &region, NULL, line_outline_callback, &info);
-  r_search (Layer->arc_tree,  &region, NULL, arc_outline_callback, &info);
+  if (found_outline)
+    {
+      r_search (Layer->line_tree, &region, NULL, line_outline_callback, &info);
+      r_search (Layer->arc_tree,  &region, NULL, arc_outline_callback, &info);
+    }
 
   if (include_holes)
     {
